--- a/data/com.github.robertsanseries.ciano.appdata.xml.in	2024-09-22 11:49:28.181541667 -0500
+++ b/data/com.github.robertsanseries.ciano.appdata.xml.in	2024-09-22 14:23:21.412870482 -0500
@@ -2,6 +2,7 @@
 <!-- Copyright 2017 Robert San <robertsanseries@gmail.io> -->
 <component type="desktop-application">
     <id>com.github.robertsanseries.ciano</id>
+    <launchable type="desktop-id">com.github.robertsanseries.ciano.desktop</launchable>
     <metadata_license>CC0-1.0</metadata_license>
     <project_license>GPL-3.0+</project_license>
     <name>Ciano</name>
@@ -10,6 +11,7 @@
     <provides>
         <binary>com.github.robertsanseries.ciano</binary>
     </provides>
+    <translation type="gettext">ciano</translation>
     <screenshots>
         <screenshot type="default">
             <image>https://raw.githubusercontent.com/robertsanseries/ciano/master/data/screenshot.png</image>
@@ -131,6 +133,10 @@
             </description>
         </release>
     </releases>
+    <categories>
+        <category>Utility</category>
+        <category>AudioVideo</category>
+    </categories>
     <content_rating type="oars-1.1">
         <content_attribute id="violence-cartoon">none</content_attribute>
         <content_attribute id="violence-fantasy">none</content_attribute>
